app/src/main/java/com/example/ithaca_transit_android_v2/MainActivity.kt

```diff
@@ -1,50 +1,134 @@
+                }
+            }
+
+            emitter.setCancellable {
+                search_input.removeTextChangedListener(watcher)
+                search_input.setOnFocusChangeListener(null)
+            }
+        }
+        return obs.startWith(SearchLaunchState())
+    }
+
+    fun initSearchView() {
+        val observable = createSearchObservable()
+
+        disposable = observable
+            .observeOn(Schedulers.io())
+            .map { state ->
+                if (state is InitSearchState) {
+                    val locations = NetworkUtils().getSearchedLocations(state.searchText)
+                    InitLocationsSearchState(state.searchText, locations)
+
+                } else {
+                    state
+                }
+            }
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe ({ state ->
+                when (state) {
+                    is SearchLaunchState -> {
+                        search_empty_state.visibility = View.GONE
+                        search_locations_state.visibility = View.GONE
+                    }
+                    is EmptyInitClickState -> {
+                        search_locations_state.visibility = View.GONE
+                        search_empty_state.visibility = View.VISIBLE
+                    }
+                    is InitLocationsSearchState -> {
+                        search_empty_state.visibility = View.GONE
+                        search_locations_state.visibility = View.VISIBLE
+                        if(state.searchedLocations!!.size > 0 || search_input.text.isEmpty()) {
+                            mSearchAdapter!!.swapItems(state.searchedLocations)
+                        }
+                    }
+                }
+            }, {error -> Log.e("An Error Occurred", error.toString())})
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        // ======== FOR TESTING ONLY ========
-        val end = Coordinate(42.444971674516864, -76.48098092526197)
-        // val uid = "E4A0256E-5865-4E9F-8A5A-33747CAC7EBF"
-        val time = 1574292741.0
-        val destinationName = "Bill & Melinda Gates Hall"
-        val start = Coordinate(42.44717985041025, -76.48551732274225)
-        val arriveBy = false
-
-        // TODO (lesley): just for testing purposes - replace with rxjava
-        runBlocking {
-            val deferred = CoroutineScope(Dispatchers.IO).async {
-                NetworkUtils().getRouteOptions(start, end, time, arriveBy, destinationName)
-            }.await()
-
-            // Printing out [deferred] in log for testing
-            printLongLog(deferred.toString())
+        mSearchAdapter = SearchViewAdapter(this, mSearchLocations)
+        locations_list.adapter = mSearchAdapter
+        locations_list.setOnItemClickListener{parent, view, position, id ->
```

Comment:
should invoke/emit a `MapState` so that the map changes. it's outside the scope of this pr so feel free to leave a `TODO`